### ISSUE TITLE
Introduce exports/require or similar somewhere between Jest and Data Groups

### ISSUE THREAD

**Issue by illicitonion** [CodeYourFuture/curriculum#1062]
Data Groups has instructions that assume you're writing separate implementation and test files (e.g. in https://programming.codeyourfuture.io/data-groups/sprints/1/prep/#mean)

But until this point we've only ever written things in one file, and never used something like `import`/`require`/`exports`.

For now I suspect trainees will just implement everything in the `.test.js` files they create (and maybe even not notice that they're making other files).

But the coursework for this sprint (https://github.com/CodeYourFuture/Module-Data-Groups/tree/main/Sprint-1) has some separate implementations and then starts just assuming you'll know how to set things up from scratch.

For now I'm going to add a little scaffolding, but we should introduce these concepts (maybe in the Jest sprint following learning Jest?) so our trainees can start practicing them.

**Comment by SallyMcGrath** [comment]
We set it up in the components workshops

**Comment by illicitonion** [comment]
A few risks here:
1. Workshops are optional, both in terms of "who comes to class and does them" and in terms of "the workshop block lists multiple possible workshops" (and on top of that, components 2 has a prereq on components 1, so the correct workshop needs to have been selected in _both_ sprints). We should work out how to make both of these workshops mandatory, if we're going to rely on knowledge from them.
2. That workshop has a _lot_ of new concepts, content, and goals. It's likely that even if that workshop is attempted, many trainees won't get through enough of the content to learn this particular concept (e.g. they may end up spending an hour learning about git branching workflows).

I think it's worth throwing in one short 10 minute block on this just in case :)

**Comment by SallyMcGrath** [comment]
Yeah - it's really hard to know how to get people to do things, I agree. Tangentially, we have this big problem with nobody looking at the lesson plan before showing up. We used to try to fix this with a planning meeting, but then the Proms all drove themselves into the ground trying to get people to go to the meeting so the solution was more burdensome than the problem.

I don't have a good answer here. When I was a volunteer and I was leading the module, I prepared for some hours before class. When I wasn't leading, I often didn't prepare, (but I did skim as I was joining). A leader is helpful to take responsibility for knowing what is happening on the day. But what if we don't have them or not enough of them -- we end up with one person being lumped with everything, which typically happened by the course mid point.

**Comment by illicitonion** [comment]
I agree with those problems of class leadership and preparation, but I feel like this is a slightly different problem.

Currently we don't view workshops to be a required element of the course. So we shouldn't rely on them for content in the future of the course. They can be great for practicing, re-enforcing, opening up new ideas on the side, etc, but if some content from them is _required_ knowledge, we should _also_ teach it somewhere else (which a trainee may breeze through really quickly if they happen to have already learnt the content through a workshop).

(Alternatively we could try to make workshops required, but I think that's a much bigger and harder change...)
